File: 프로그래머스/unrated/181833. 특별한 이차원 배열 1/특별한 이차원 배열 1.py
def solution(n):
    # answer 에 n*n 배열을 생성하고 기본값으로 0 을 삽입
    answer= [[0 for _ in range(n)] for _ in range(n)]
    
    # 배열의 모든 값 탐색
    for i in range(n):
        for j in range(n):
            # i = j 일 때 배열에 1 삽입
            if i == j:
                answer[i][j] = 1
    return answer

File: 프로그래머스/unrated/181833. 특별한 이차원 배열 1/특별한 이차원 배열 1.js
function solution(n) {
    const result = [];
    for(let i=0;i<n;i++){
        let arr = [];
        for(let j=0;j<n;j++){
            i===j?arr.push(1):arr.push(0);
        }
        result.push(arr);
    }
    return result;
}